src/app/components/map.tsx

```diff
@@ -0,0 +1,91 @@
+"use client";
+
+import "mapbox-gl/dist/mapbox-gl.css";
+import mapboxgl from "mapbox-gl";
+import { useEffect, useRef } from "react";
+import { NearbySightingsResponse } from "../recent/nearby/page";
+
+export default function MapComponent({
+  onDragEnd,
+  nearbySightings,
+}: {
+  nearbySightings: NearbySightingsResponse | undefined;
+  onDragEnd: (longitude: number, latitude: number) => void;
+}) {
+  const mapContainer = useRef<HTMLDivElement>(null);
+  const mapRef = useRef<mapboxgl.Map | null>(null);
+  const markersRef = useRef<mapboxgl.Marker[]>([]);
+
+  if (!process.env.NEXT_PUBLIC_MAPBOX_API_TOKEN) {
+    throw new Error("Missing mapBox API token");
+  }
+
+  useEffect(() => {
+    const accessToken = process.env.NEXT_PUBLIC_MAPBOX_API_TOKEN;
+    mapboxgl.accessToken = `${accessToken}`;
+    if (mapContainer.current) {
+      const map = new mapboxgl.Map({
+        container: mapContainer.current,
+        style: "mapbox://styles/mapbox/outdoors-v12",
+        center: [-1.688074, 53.185335], // make center the most recent sighted bird or something
+        zoom: 10,
+        maxZoom: 15,
+        attributionControl: false,
+      });
+
+      mapRef.current = map;
+
+      map.addControl(new mapboxgl.NavigationControl(), "top-left");
+
+      let geolocate = new mapboxgl.GeolocateControl({
+        positionOptions: {
+          enableHighAccuracy: true,
+        },
+        trackUserLocation: true,
+      });
+
+      map.addControl(geolocate);
+
+      const handleMoveEvents = () => {
+        const longitude = map.getCenter().lng;
+        const latitude = map.getCenter().lat;
+        onDragEnd(longitude, latitude);
+      };
+
+      geolocate.on("geolocate", () => {
+        handleMoveEvents();
+      });
+      map.on("dragend", () => {
+        handleMoveEvents();
+      });
+
+      return () => map.remove();
+    }
+  }, []);
```

Comment:
How could a user re-define onDragEnd?

Otherwise sounds good, will leave it for now and I've put a ticket on the board.

Reply:
Ah apologies I wasn't especially clear there at all. 

I meant if any consumer of this component (as in another engineer) redefines what they pass to `onDragEnd` during the lifetime of the app.

This could happens if they pass a function to it that is re-defined when their component is rendered, which is most in-component functions. 

Perhaps their function is relying on some state within their component. When that state changes the component re-render and the function is redefined, however because this effect doesn't have it as a dependency it won't re-run. So now their function is using different state to the function  this effect has stored.

Hopefully that makes sense 😅. If not we can talk about it in a call some time!